yt/chyt/controller/internal/jupyt/controller.go

```diff
@@ -193,6 +243,54 @@ func (c *Controller) appendConfigs(ctx context.Context, oplet *strawberry.Oplet,
 	return nil
 }
 
+func (c *Controller) GetScalerTarget(ctx context.Context, opletInfo strawberry.OpletInfoForScaler) (*strawberry.ScalerTarget, error) {
+	endpointInfo, err := getEndpoint(ctx, c.ytc, opletInfo.OperationID)
+	if err != nil {
+		return nil, err
+	}
+
+	lastActivityRelUrl := c.config.LastActivityURLPathOrDefault()
+	lastActivityUrl := fmt.Sprintf("%s/%s", endpointInfo.Address, lastActivityRelUrl)
+	resp, err := http.Get(lastActivityUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("jupyter responded with error: %s", body)
+	}
+	var jupyterResponse struct {
+		LastActivity time.Time `json:"last_activity"`
+	}
+	err = json.NewDecoder(bytes.NewReader(body)).Decode(&jupyterResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	speclet := opletInfo.ControllerSpeclet.(Speclet)
+	sinceLastActivity := time.Now().Sub(jupyterResponse.LastActivity)
+	if speclet.EnableIdleSuspension && sinceLastActivity > speclet.IdleTimeoutOrDefault() {
```

Comment:
If "idle suspension" is disabled we don't need to get endpoint/make the http request - it's a waste of time